### ISSUE TITLE
never_loop does not pick up on continue in a loop

### ISSUE THREAD

**Issue by sergmister** [rust-lang/rust-clippy#7415]
Lint name: clippy::never_loop

This works:
```rust
loop {
    if true {
        continue;
    }
    break;
}
```
even like this:
```rust
if false {
    continue;
}
```

This gives error: this loop never actually loops
`#[deny(clippy::never_loop)]` on by default
```rust
loop {
    for _ in 0..1 {
        continue;
    }
    break;
}
```

Continue could / will be called in a loop so this lint should not be shown. Clippy can detect and omit this lint with `continue` inside an if block but not within a loop.

### Meta

clippy 0.1.54 (868c702 2021-06-30)

rustc 1.55.0-nightly (868c702d0 2021-06-30)
binary: rustc
commit-hash: 868c702d0c9a471a28fb55f0148eb1e3e8b1dcc5
commit-date: 2021-06-30
host: x86_64-unknown-linux-gnu
release: 1.55.0-nightly
LLVM version: 12.0.1

**Comment by camsteffen** [comment]
Can you reproduce this in a rust playground? I wasn't able to.

**Comment by sergmister** [comment]
> Can you reproduce this in a rust playground? I wasn't able to.

[https://play.rust-lang.org/?version=nightly&mode=debug&edition=2018&gist=834ba9d28f197854ebe0cac151a167dd](https://play.rust-lang.org/?version=nightly&mode=debug&edition=2018&gist=834ba9d28f197854ebe0cac151a167dd)

**Comment by camsteffen** [comment]
Oh I was using your "this works" example 🤦‍♂️

**Comment by HKalbasi** [comment]
I don't understand this. Outer loop isn't a loop actually (continue belongs to `for` loop) and it is correctly linted. So what is the problem?

**Comment by camsteffen** [comment]
Huh you're right.
